src/e2e-test/test-rule-functions.js

```diff
@@ -0,0 +1,50 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+var testUtils = require('./testUtils');
+var fhirModule = require('fhir');
+var fhir = new fhirModule.Fhir();
+
+var fhirR4Validation =  function(resJson){
```

Comment:
We have this rule, but neither of the current rule sets use it. Should it be used in a test? It seems valuable for template verification.

Reply:
Yeah. I didn't use it in all rule sets because all the current conversion results can't pass the test now. However, I have some unit-tests in test-rule-functions.spec.js.

Reply:
It might be tough to reach zero validation error (to enable this test) and stay there consistently. Is it possible to add upper limit on validation failures for this? So that we can at least start using this test with some limit and ensure subsequent changes don't increase the validation failure count.

Reply:
We can use this rule by modifying templates slightly instead of adding upper limits because most of them just lack the necessary properties, such as test case `dataFile: 'C-CDA_R2-1_CCD.xml.cda', templateFile: 'ccd.hbs'`, only these two errors need to be corrected:
```
{
        "location": "Coverage.beneficiary",
        "resourceId": "ebd801b5-9113-3479-b35b-71002aae6c28",
        "severity": "error",
        "message": "Missing property"
},
{
        "location": "Coverage.payor",
        "resourceId": "ebd801b5-9113-3479-b35b-71002aae6c28",
        "severity": "error",
        "message": "Missing property"
}
```

Reply:
It might work for small errors, but doesn't seem sustainable as we add more stuff. For example, when we start adding new data types, there are always a lot of errors which get fixed slowly over a course of time. It would be great if we can add some tests in that period and improve on those test cases with time. we can think of it as code coverage numbers. 
How about you create a new user story for this change, which can be done separately?

Reply:
Yes, I created a new story about Templates tests and updates using e2e test framework. I will try to use this rule and we can have more detailed discussion later.